src/olympia/addons/serializers.py

```diff
@@ -493,6 +496,25 @@ def validate_upload(self, value):
             raise Conflict({self.field_name: exc.messages})
         return value
 
+    def validate_is_disabled(self, disable):
+        # The Version.is_user_disabled setter just ignores any change it doesn't like,
+        # but we'd like to raise instead if the value wouldn't be accepted.
+        if disable and (version := self.instance):
+            if (
+                not version.is_user_disabled
```

Comment:
Do we need this part of the condition ? It seems fine but I don't understand why it's there.

Reply:
We do - to only raise when there is an attempt to change disabled value (i.e. it wasn't already user disabled).
I used `version.is_user_disabled` to keep the logic about whether the version is user disabled in one place but, looking now, this could be replaced with:
`if version.file.status == amo.STATUS_DISABLED and version.file.original_status == amo.STATUS_NULL:`
if that's clearer?

Reply:
But if the `file.status` is `amo.STATUS_DISABLED`, you shouldn't be able to flip `is_user_disabled`, right ? So that part of the condition seems superfluous.

Reply:
Ok, I see it now.

I was wondering about modifiying the version in general when it's `STATUS_DISABLED` already. It is currently possible in devhub, so I guess that's fine...